### ISSUE TITLE
[Bug]: TestSelectWithMultiPartition

### ISSUE THREAD

**Issue by daviszhen** [matrixorigin/matrixone#19257]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Branch Name

main

### Commit ID

lastest

### Other Environment Information

```Markdown
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

```

+--------------------------------------------------------------------------------------------------------------------+
| Failed UT Packages                                                                                                 |
+---------------------------------------------------+----------------------+----------------------+--------+---------+
| PACKAGE                                           | START_TIME           | END_TIME             | RESULT | ELAPSED |
+---------------------------------------------------+----------------------+----------------------+--------+---------+
| github.com/matrixorigin/matrixone/pkg/tests/shard | 20[24](https://github.com/matrixorigin/matrixone/actions/runs/11302116760/job/31437505529?pr=19249#step:9:25)-10-12T03:11:52Z | 2024-10-12T03:14:35Z | fail   | 162.81  |
+---------------------------------------------------+----------------------+----------------------+--------+---------+
============================================
[TEST]: TestSelectWithMultiPartition
[START_AT]: 2024-10-12T03:12:53Z
[END_AT]: 2024-10-12T03:12:59Z
[RESULT]: fail
[ELAPSED]: 5.78
[OUTPUTS]:
=== RUN   TestSelectWithMultiPartition
2024/10/12 03:12:53.576708 +0000 INFO tn-service.lockservice.lockservice.allocator lo
```

### Expected Behavior

no failure

### Steps to Reproduce

```Markdown
https://github.com/matrixorigin/matrixone/actions/runs/11302116760/job/31437505529?pr=19249
```

**Comment by sukki37** [comment]
[job-logs.txt.zip](https://github.com/user-attachments/files/17359986/job-logs.txt.zip)

```
2024-10-12T03:33:59.3114298Z 2024/10/12 03:12:59.352293 +0000 INFO compile/sql_executor.go:359 sql_executor exec {"sql": "select value from TestSelectWithMultiPartition", "txn-id": "daab5c42492d6b6917fd95f8a8c79e1c", "duration": "7.917977ms", "BatchSize": 2, "AffectedRows": 0}
2024-10-12T03:33:59.3115505Z     partition_based_test.go:552: 
2024-10-12T03:33:59.3116360Z         	Error Trace:	/home/runner/work/matrixone/matrixone/pkg/tests/shard/partition_based_test.go:552
2024-10-12T03:33:59.3117727Z         	            				/home/runner/work/matrixone/matrixone/pkg/tests/shard/partition_based_test.go:323
2024-10-12T03:33:59.3118931Z         	            				/home/runner/work/matrixone/matrixone/pkg/tests/shard/shard_test.go:108
2024-10-12T03:33:59.3120067Z         	            				/home/runner/work/matrixone/matrixone/pkg/tests/shard/shard_test.go:40
2024-10-12T03:33:59.3121301Z         	            				/home/runner/work/matrixone/matrixone/pkg/tests/shard/partition_based_test.go:291
2024-10-12T03:33:59.3121936Z         	Error:      	Not equal: 
2024-10-12T03:33:59.3122450Z         	            	expected: []int{1, 11, 21}
2024-10-12T03:33:59.3122975Z         	            	actual  : []int{1, 11}
2024-10-12T03:33:59.3123501Z         	            	
2024-10-12T03:33:59.3123846Z         	            	Diff:
2024-10-12T03:33:59.3124284Z         	            	--- Expected
2024-10-12T03:33:59.3124697Z         	            	+++ Actual
2024-10-12T03:33:59.3125194Z         	            	@@ -1,5 +1,4 @@
2024-10-12T03:33:59.3125826Z         	            	-([]int) (len=3) {
2024-10-12T03:33:59.3126310Z         	            	+([]int) (len=2) {
2024-10-12T03:33:59.3126739Z         	            	  (int) 1,
2024-10-12T03:33:59.3127169Z         	            	- (int) 11,
2024-10-12T03:33:59.3127601Z         	            	- (int) 21
2024-10-12T03:33:59.3128068Z         	            	+ (int) 11
2024-10-12T03:33:59.3128423Z         	            	 }
2024-10-12T03:33:59.3128836Z         	Test:       	TestSelectWithMultiPartition
2024-10-12T03:33:59.3129349Z --- FAIL: TestSelectWithMultiPartition (5.78s)
```

**Comment by triump2020** [comment]
replicated with #19202

**Comment by triump2020** [comment]
Pls ref to #19202

**Comment by zhangxu19830126** [comment]
fixed by #19530
